tests/evm/test_begin_tx.py

```diff
@@ -0,0 +1,113 @@
+from zkevm_specs.evm import (
+    main, ExecutionResult, CoreState, CallState, Step,
+    Tables, TxTableTag, CallContextTag, RWTableTag,
+)
+from zkevm_specs.util import fp_inv, linear_combine, keccak256, EMPTY_CODE_HASH
+
+
+# TODO: Parametrize r, tables
+def test_begin_tx():
+    r = 1
+    bytecode = bytes.fromhex('00')
```

Comment:
I understand begin_tx is a flag we set in circuit side as a placeholder, there is no explicitly care in evm , if so the call state and others  i.e. allocations . should be empty at this step , right ?

Reply:
Yeah, the `call_state` will not be used and no opcode will be fetched. However, we still need allocation to verify a bunch of stuff, like verifying if it could buy intrinsic gas, transfer ether, callee has code or not, etc...

Reply:
Do we need to assign an opcode id to BEGIN_TX?

Reply:
The idea here is try to define our own finite state machine to emulate EVM, because the opcode in EVM cost very differently between each other, and the error cases make even simple opcode complicated. So I try to enumerate all kind of possible execution result of a step, and repeat such step to emulate EVM.

Then `BEGIN_TX` is just one kind of execution result which doesn't consume opcode, there might be some pre-compiles do so.